src/main/java/org/isf/telemetry/gui/TelemetryEdit.java

```diff
@@ -0,0 +1,447 @@
+		confirmButton.addActionListener(buildConfirmationActionListener(checkboxes, telemetryManager, telemetryUtils));
+		return confirmButton;
+	}
+
+	private JButton buildAskMeLaterButton() {
+		JButton cancelButton = new JButton(MessageBundle.getMessage(KEY_TELEMETRY_BUTTON_LABEL_ASK_LATER));
+		cancelButton.addActionListener(buildAskMeLaterButtonActionListener(telemetryManager));
+		return cancelButton;
+	}
+
+	private JButton buildDisableNeverAskButton() {
+		JButton cancelButton = new JButton(MessageBundle.getMessage(KEY_TELEMETRY_BUTTON_LABEL_DISABLE_NEVER_ASK));
+		cancelButton.addActionListener(buildDisableNeverAskButtonActionListener(telemetryManager));
+		return cancelButton;
+	}
+
+	private JButton buildDisableButton() {
+		JButton disableButton = new JButton(MessageBundle.getMessage(KEY_TELEMETRY_BUTTON_LABEL_DISABLE));
+		disableButton.addActionListener(buildDisableNeverAskButtonActionListener(telemetryManager));
+		return disableButton;
+	}
+
+	private JButton buildCloseButton() {
+		JButton closeButton = new JButton(MessageBundle.getMessage(KEY_TELEMETRY_BUTTON_LABEL_CLOSE));
+		closeButton.addActionListener(closeButtonActionListener(telemetryManager));
+		return closeButton;
+	}
+
+	/**
+	 * We could load checkboxes information from somewhere (properties file or from
+	 * other strange places)
+	 * 
+	 * @return
+	 */
+	private List<CheckBoxWrapper> buildPermissionCheckboxes(ApplicationContext applicationContext,
+					Map<String, Boolean> consentMap) {
+
+		Map<String, AbstractDataCollector> checkboxContractMap = applicationContext.getBeansOfType(AbstractDataCollector.class);
+		List<AbstractDataCollector> checkboxContractList = new ArrayList<>(checkboxContractMap.values());
+		Collections.sort(checkboxContractList, AnnotationAwareOrderComparator.INSTANCE);
+
+		List<CheckBoxWrapper> result = new ArrayList<>();
+
+		int[] i = { 0 };
+		checkboxContractList.forEach(springCheckboxConfigurationBean -> {
+			JCheckBox chb = new JCheckBox(springCheckboxConfigurationBean.getDescription(),
+							springCheckboxConfigurationBean.isSelected(consentMap));
+			CheckBoxWrapper wrapper = new CheckBoxWrapper();
+			wrapper.setCheckbox(chb);
+			wrapper.setId(springCheckboxConfigurationBean.getId());
+			wrapper.setOrder(Integer.valueOf(i[0]++));
+			if (springCheckboxConfigurationBean.getId().equals("TEL_ID")) {
+				// Compulsory Collector
+				wrapper.getCheckbox().setSelected(true);
+				wrapper.getCheckbox().setEnabled(false);
+			}
+			result.add(wrapper);
+		});
+
+		return result;
+	}
+
+	/**
+	 * Action for confirmation button: inserts for the first time in the telemetry
+	 * table or updates the existing row
+	 * 
+	 * @param permissions
+	 * @param telemetryManager
+	 * @return
+	 */
+	private ActionListener buildConfirmationActionListener(List<CheckBoxWrapper> checkboxes,
+					TelemetryManager telemetryManager, TelemetryUtils telemetryUtils) {
+		JPanel panel = this.panel;
```

Comment:
why is this defined here if it is only used in line 303?  Is it even needed; doesn't the dialog have a default frame it uses?

Reply:
it was a typo, fixed.